tests/usm/usm_fill_memset_memcpy.cpp

```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Checks functions fill, memset, and memcpy for Unified Shared Memory API.
+//
+*******************************************************************************/
+
+#include "../common/common.h"
+#include <cstring>
+
+#define TEST_NAME usm_fill_memset_memcpy
+
+namespace TEST_NAMESPACE {
+using namespace sycl_cts;
+
+/** Test instance
+ */
+class TEST_NAME : public sycl_cts::util::test_base {
+ public:
+  /** return information about this test
+   */
+  void get_info(test_base::info &out) const override {
+    set_test_info(out, TOSTRING(TEST_NAME), TEST_FILE);
+  }
+
+  /** execute the test
+   */
+  void run(util::logger &log) override {
+    try {
+      auto q = util::get_cts_object::queue();
+      if (!q.get_device().has(sycl::aspect::usm_shared_allocations)) {
+        log.note(
+            "Device does not support accessing to unified shared memory "
+            "allocation");
+        return;
+      }
+
+      constexpr auto count{2};
+      constexpr auto size{sizeof(int) * count};
+      constexpr auto value_for_filling{1};
+      constexpr auto value_for_first_element_overwriting{10};
+
+      auto deleter = [&q](int *data) { sycl::free(data, q.get_context()); };
+      std::unique_ptr<int, decltype(deleter)> src(
+          sycl::malloc_shared<int>(size, q.get_device(), q.get_context()),
+          deleter);
+      std::unique_ptr<int, decltype(deleter)> output(
```

Comment:
```suggestion
      std::unique_ptr src(
          sycl::malloc_shared<int>(size, q.get_device(), q.get_context()),
          deleter);
      std::unique_ptr output(
```
thanks to C++17 CTAD

Reply:
@keryell How will it know that's an `unique_ptr<int>` or `unique_ptr<int[]>` though?

Reply:
@keryell  I've tried to apply your suggestion but got a compilation error: "usm_fill_memset_memcpy.cpp(45,23): error: no viable constructor or deduction guide for deduction of template arguments of 'unique_ptr'
      std::unique_ptr src("

Reply:
Hi Ronan. I've tried to apply your suggestion but got a compilation error. I have no other ideas for changing this part of code.

Reply:
Curious. It looks like this is by design according to https://en.cppreference.com/w/cpp/memory/unique_ptr/unique_ptr : "The program is ill-formed if this constructor is selected by class template argument deduction."
This does not work even for simple cases: https://godbolt.org/z/9GeqbTd8h
So, keep it like you did. Thanks for having tried.

Reply:
@keryell  Do I understand correctly that you approve this PR?

Reply:
Oh yes! Thank you for reminding this. Sorry for the distraction...